internal/db/testing.go

```diff
@@ -30,7 +31,8 @@ func TestSetup(t *testing.T, dialect string, opt ...TestOption) (*gorm.DB, strin
 
 	switch opts.withTestDatabaseUrl {
 	case "":
-		cleanup, url, _, err = StartDbInDocker(dialect)
+		opts := docker.WithContainerImage(dialect)
```

Comment:
Note that our "dialect" is actually `"postgres"`. This means that the net effect of passing `dialect` straight through will be that it will not have any colons, so will split into a db and tag of `postgres:latest` instead of what we want, which is `postgres:11`.

The disconnect here is that the dialect is referring to the SQL language understood, not a particular version of a server that speaks it. (You could for instance have a dialect of `postgres` and actually use a `cockroachdb` image.) In fact in `manager_test.go` above (yes, I'm reviewing upwards), you can see that issue -- you're creating a dialect of `postgres` which is correct for `sql.Open` but will result in the wrong image (`postgres:latest`).

I think you need to re-add dialect as a parameter in the places where you replaced it with the options, and use them in the appropriate and distinct ways. They're separate things.